coordinator/src/test/scala/filodb.coordinator/queryplanner/SingleClusterPlannerSpec.scala

```diff
@@ -449,13 +450,205 @@ class SingleClusterPlannerSpec extends AnyFunSpec with Matchers with ScalaFuture
+      // Should pushdown inner join.
+      ("""sum(foo{job="baz"} + on(job, app) bat{job="baz"})""",
+        """T~AggregatePresenter(aggrOp=Sum, aggrParams=List(), rangeParams=RangeParams(20000,100,30000))
+          |-E~LocalPartitionReduceAggregateExec(aggrOp=Sum, aggrParams=List()) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-852082083],raw)
+          |--T~AggregateMapReduce(aggrOp=Sum, aggrParams=List(), without=List(), by=List())
+          |---E~BinaryJoinExec(binaryOp=ADD, on=List(job, app), ignoring=List()) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-852082083],raw)
+          |----T~PeriodicSamplesMapper(start=20000000, step=100000, end=30000000, window=None, functionId=None, rawSource=true, offsetMs=None)
+          |-----E~MultiSchemaPartitionsExec(dataset=timeseries, shard=1, chunkMethod=TimeRangeChunkScan(19700000,30000000), filters=List(ColumnFilter(job,Equals(baz)), ColumnFilter(__name__,Equals(foo))), colName=None, schema=None) on InProcessPlanDispatcher(filodb.core.query.QueryConfig@7c3e4b1a)
+          |----T~PeriodicSamplesMapper(start=20000000, step=100000, end=30000000, window=None, functionId=None, rawSource=true, offsetMs=None)
+          |-----E~MultiSchemaPartitionsExec(dataset=timeseries, shard=1, chunkMethod=TimeRangeChunkScan(19700000,30000000), filters=List(ColumnFilter(job,Equals(baz)), ColumnFilter(__name__,Equals(bat))), colName=None, schema=None) on InProcessPlanDispatcher(filodb.core.query.QueryConfig@7c3e4b1a)
+          |--T~AggregateMapReduce(aggrOp=Sum, aggrParams=List(), without=List(), by=List())
+          |---E~BinaryJoinExec(binaryOp=ADD, on=List(job, app), ignoring=List()) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-852082083],raw)
+          |----T~PeriodicSamplesMapper(start=20000000, step=100000, end=30000000, window=None, functionId=None, rawSource=true, offsetMs=None)
+          |-----E~MultiSchemaPartitionsExec(dataset=timeseries, shard=17, chunkMethod=TimeRangeChunkScan(19700000,30000000), filters=List(ColumnFilter(job,Equals(baz)), ColumnFilter(__name__,Equals(foo))), colName=None, schema=None) on InProcessPlanDispatcher(filodb.core.query.QueryConfig@7c3e4b1a)
+          |----T~PeriodicSamplesMapper(start=20000000, step=100000, end=30000000, window=None, functionId=None, rawSource=true, offsetMs=None)
+          |-----E~MultiSchemaPartitionsExec(dataset=timeseries, shard=17, chunkMethod=TimeRangeChunkScan(19700000,30000000), filters=List(ColumnFilter(job,Equals(baz)), ColumnFilter(__name__,Equals(bat))), colName=None, schema=None) on InProcessPlanDispatcher(filodb.core.query.QueryConfig@7c3e4b1a)""".stripMargin),
+      // scalars should not prevent pushdown
+      ("""foo{job="baz"} + 5""",
```

Comment:
Excellent that we cover this.
Do we also do pushdown for query such as `foo or vector(0)` ? This is a very common query that is issued today, and would be nice to optimize. If yes, please add a test case. If no, let us add a todo and a ticket to cover this in a separate PR. I dont want for us to further bloat this one. 🙂

Reply:
Good catch. I've added a case in `getBinaryPushdownShards` to prevent `VectorPlan` pushdowns. The reasoning is:
```
      // VectorPlans can't currently be pushed down. Consider:
      //     foo{...} or vector(0)
      //   If foo{...} is sharded with spread=1, but both shards contain no foo{...} data,
      //   then both pushed-down plans will yield vector(0). These will be concatenated via a DistConcatExec.
```

Reply:
Ticket is created.

Reply:
Thank, lets create a unit test that ensures that push-down does not happen for vector(0) binary joins.

Reply:
Done (glad you suggested this-- it caught a bug).